lua/ttt2/libraries/targetid.lua

```diff
@@ -0,0 +1,558 @@
+---
+-- @module TargetID
+-- @desc Handling operations to display the TargetID
+-- @author Mineotopia
+
+if SERVER then
+    AddCSLuaFile()
+    return
+end
+
+targetid = targetid or {}
+
+-- Global to local variables
+local bIsInitialized = false
+local ParT
+local TryT
+
+-- Variables for calculations
+local MAX_TRACE_LENGTH = math.sqrt(3) * 32768
+
+-- Key Parameters for doors
+local key_params = {}
+
+-- Convars for targetid
+ local cvDeteOnlyConfirm = GetConVar("ttt2_confirm_detective_only")
+ local cvDeteOnlyInspect = GetConVar("ttt2_inspect_detective_only")
+
+-- Materials for targetid
+local materialTButton = Material("vgui/ttt/tid/tid_big_tbutton_pointer")
+local materialRing = Material("effects/select_ring")
+local materialRoleUnknown = Material("vgui/ttt/tid/tid_big_role_not_known")
+local materialDisguised = Material("vgui/ttt/perks/hud_disguiser.png")
+local materialCorpse = Material("vgui/ttt/tid/tid_big_corpse")
+local materialCredits = Material("vgui/ttt/tid/tid_credits")
+local materialDetective = Material("vgui/ttt/tid/tid_detective")
+local materialLocked = Material("vgui/ttt/tid/tid_locked")
+local materialAutoClose = Material("vgui/ttt/tid/tid_auto_close")
+local materialDoor = Material("vgui/ttt/tid/tid_big_door")
+local materialDestructible = Material("vgui/ttt/tid/tid_destructible")
+local materialDNATargetID = Material("vgui/ttt/dnascanner/dna_hud")
+
+---
+-- This function is a temporary fix for local variables in use of other libraries, that are not yet initialized
+function targetid.initialize()
+    if not bIsInitialized then
+        bIsInitialized = true
+        ParT = LANG.GetParamTranslation
+        TryT = LANG.TryTranslation
+        key_params = {
+            usekey = Key("+use", "USE"),
+            walkkey = Key("+walk", "WALK")
+        }
+    end
+end
```

Comment:
This is the current Initialization with libraries that are not yet loaded, when this library is loaded.
A better solution should be searched for.